File: amp-cli/src/commands/reindex.ts
import { Command } from "commander";
import { reindexStore } from "../core/index.js";
import { requireStore } from "../core/store.js";
import { kv, success } from "../utils/ui.js";

export function registerReindexCommand(program: Command): void {
  program
    .command("reindex")
    .description("Rebuild index/keywords.json and index/graph.json from the current nodes")
    .action(() => {
      const paths = requireStore(process.cwd());
      const { nodeCount, graph } = reindexStore(paths);
      const linkCount = Object.values(graph.nodes).reduce((sum, n) => sum + n.outgoing.length, 0);

      console.log(success("Reindexed store"));
      console.log(kv("nodes", String(nodeCount)));
      console.log(kv("links", String(linkCount)));
    });
}

File: amp-python/src/amp/commands/reindex.py
from __future__ import annotations

import click

from ._common import console, get_store, handle_amp_errors


@click.command()
@click.pass_context
@handle_amp_errors
def reindex_cmd(ctx: click.Context) -> None:
    """Rebuild index/keywords.json and index/graph.json."""
    store = get_store(ctx.obj.get("path"))
    keywords, graph = store.reindex()
    console.print(
        f"[green]Reindexed[/green] {len(keywords)} node(s) — "
        f"{sum(len(v) for v in graph['outgoing'].values())} outgoing link(s)."
    )
